frappe/core/doctype/prepared_report/prepared_report.py

```diff
@@ -56,13 +55,35 @@ def get_prepared_data(self, with_file_name=False):
 			return gzip_decompress(attached_file.get_content())
 
 
-def generate_report(prepared_report):
-	update_job_id(prepared_report, get_current_job().id)
+def generate_report(prepared_report_name):
+	prepared_report_instance = frappe.get_doc("Prepared Report", prepared_report_name)
+	report = frappe.get_doc("Report", prepared_report_instance.report_name)
+
+	filters_month_diff = month_diff(
```

Comment:
All this logic should not live in.


Framework can provide hook/abstraction for reports that can be "chunked" how to chunk and what size should be chunked etc should be left to report. Monthly chunks appear to be a common pattern now, but soon enough you'll have weeks, day, partition keys (like company, warehouse) etc.